### PR TITLE
feat(useLayer): allow styling border without `!` for outline

### PR DESCRIPTION
**Pull request by zguolee**

### CODE REVIEW

**Comment by zguolee** [comment]
Sorry, I was affected by the variant attribute and forgot to be able to add it directly through the `class`.🥲 Thank you for this project, very awesome, I also learned a lot.

**Comment by jd-solanki** [comment]
> Sorry, I was affected by the variant attribute and forgot to be able to add it directly through the `class`.🥲 Thank you for this project, very awesome, I also learned a lot.

Thanks for the PR anyways. I can't express how happy I am that you gave PR ❤️ 

Happy coding 🙌🏻

**Comment by jd-solanki** [comment]
While driving back home, I thought your PR is actually useful in other terms. We still have to add `!` (_important_) before `border-dashed`.

Moreover, we have `border border-solid` in `useLayer` which generates duplicate styles for `border-style` property.

Thanks to your PR, we can allow `border-dashed` without `!` and remove the duplicate styles.

File: packages/documentation/docs/guide/components/button.md [hunk 1]
```diff
@@ -34,6 +34,23 @@ You can use `variant="outline"` to create outlined button.
...
+<DemoButtonDashed />
+
+<template #code>
+
+<<< @/demos/button/DemoButtonDashed.vue
+
+</template>
+
+</Demo>
+
```
Comment (jd-solanki) [inline packages/documentation/docs/guide/components/button.md:53]:
```suggestion
:::details Customize `border-style` of outlined buttons
To create outlined button with different border style just add relevant class.

e.g. To create outline button with dashed border, add `border-dashed` class.
:::

```

File: packages/anu-vue/src/composables/useLayer.ts [hunk 2]
```diff
@@ -28,16 +28,24 @@ export const useLayer = () => {
     const classes: string[] = [props.states ? 'states' : '']
 
     if (props.color) {
```
Comment (jd-solanki) [inline packages/anu-vue/src/composables/useLayer.ts:30]: For `props.variant === 'outline'` condition use below:
```diff
-        classes.push(`border border-solid border-${props.color} text-${props.color} typography-title-${props.color} typography-subtitle-${props.color} typography-text-${props.color} typography-subtitle-opacity-100 typography-text-opacity-100`)
+        classes.push(`border-width-1 uno-layer-base-border-solid border-${props.color} text-${props.color} typography-title-${props.color} typography-subtitle-${props.color} typography-text-${props.color} typography-subtitle-opacity-100 typography-text-opacity-100`)
```

**Review by jd-solanki — CHANGES_REQUESTED** [review]
I want to give you credit for your first PR in this repo ❤️ 

Can you please make mentioned changes so we can merge the PR.

Thanks.

File: packages/anu-vue/src/composables/useLayer.ts (hunk 2 above)
Comment (jd-solanki) [reply packages/anu-vue/src/composables/useLayer.ts:30]: Yes 💯 

Can you please perform below changes:
```diff
        if (props.variant === 'outline')
          classes.push(`border border-${props.color}`)
        if (props.variant === 'outline')
-          classes.push(`border border-${props.color}`)
+          classes.push(`border-width-1 uno-layer-base-border-solid border-${props.color}`)
```

File: packages/anu-vue/src/composables/useLayer.ts [hunk 3]
```diff
@@ -6,7 +6,7 @@ export const useProps = (propOverrides?: Partial<ComponentObjectPropsOptions>) =
     color,
     variant: {
       type: String,
-      validator: (value: string) => ['fill', 'outline', 'light', 'text'].includes(value),
+      validator: (value: string) => ['fill', 'outline', 'dashed', 'light', 'text'].includes(value),
```
Comment (jd-solanki) [inline packages/anu-vue/src/composables/useLayer.ts:9]: We don't need `dashed` this validator as well

**Review by jd-solanki — CHANGES_REQUESTED** [review]
Apart from the two changes, everything else is correct.

Thanks for simplifying `useLayer` composable.

**Comment by zguolee** [comment]
> Out of curiosity, I wanted to ask, have you enabled "[Allow edits from maintainers](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork)"?

I checked and it was enabled.

**Review by jd-solanki — APPROVED** [review]

**Pull request merged**